crates/ruff/src/autofix/codemods.rs

```diff
@@ -0,0 +1,125 @@
+//! Interface for editing code snippets. These functions take statements or expressions as input,
+//! and return the modified code snippet as output.
+use anyhow::{bail, Result};
+use libcst_native::{
+    Codegen, CodegenState, ImportNames, ParenthesizableWhitespace, SmallStatement, Statement,
+};
+use rustpython_parser::ast::{Ranged, Stmt};
+
+use ruff_python_ast::source_code::{Locator, Stylist};
+
+use crate::cst::helpers::compose_module_path;
+use crate::cst::matchers::match_statement;
+
+/// Given an import statement, remove any imports that are specified in the `imports` iterator.
+///
+/// Returns `Ok(None)` if the statement is empty after removing the imports.
+pub(crate) fn remove_imports<'a>(
```

Comment:
Is the reason that this returns a `String` rather than an `Edit` that it may remove the entire import section?

Reply:
It's because we may need access to the modified code for some operations, and `Edit` obscures the actual modified code.